Tests/SwiftBSONTests/DocumentTests.swift

```diff
@@ -758,4 +758,28 @@ final class DocumentTests: BSONTestCase {
             expect(try BSONDocument(fromBSON: data)).to(throwError(errorType: BSONError.InvalidArgumentError.self))
         }
     }
+
+    func testWithID() throws {
```

Comment:
can we add some assertions to this test verifying that the ObjectID was added to the front of the document?